.github/workflows/ci.yaml

```diff
@@ -55,23 +60,32 @@ jobs:
         uses: actions/cache@v2
         with:
           path: ${{ steps.composer-cache.outputs.dir }}
-          key: composer-${{ runner.os }}-${{ matrix.php-version }}-${{ hashFiles('composer.*') }}
+          key: composer-${{ runner.os }}-${{ matrix.php-version }}-${{ matrix.dependencies }} }}
           restore-keys: |
-            composer-${{ runner.os }}-${{ matrix.php-version }}-${{ hashFiles('composer.*') }}-
+            composer-${{ runner.os }}-${{ matrix.php-version }}-${{ matrix.dependencies }} }}-
             composer-${{ runner.os }}-${{ matrix.php-version }}-
             composer-${{ runner.os }}-
             composer-
 
-      - name: Install dependencies
+      - name: Install locked dependencies
+        if: ${{ matrix.dependencies == 'locked' }}
+        run: |
+          composer install --no-interaction --prefer-dist --no-progress --optimize-autoloader
+
+      - name: Install lowest dependencies
+        if: ${{ matrix.dependencies == 'lowest' }}
         run: |
-          composer install --no-interaction --prefer-dist --no-progress
+          composer update --prefer-lowest --no-interaction --prefer-dist --no-progress --optimize-autoloader
+
+      - name: Install highest dependencies
+        if: ${{ matrix.dependencies == 'highest' }}
+        run: |
+          composer update --no-interaction --prefer-dist --no-progress --optimize-autoloader
 
       - name: Run tests
         shell: bash
         run: |
-          if [[ "${{ matrix.php-version }}" == '8.2' ]]; then
-            export SYMFONY_DEPRECATIONS_HELPER=max[self]=0
-          fi
+          export SYMFONY_DEPRECATIONS_HELPER=max[self]=0
```

Comment:
Maybe we can sort it out in a separate PR, but I wonder if we really want to enable the deprecations when not running the locked dependencies